O3-1.tex

```diff
@@ -279,14 +279,12 @@ \section{The LSST software stack}
 extraction of scientifically useful characteristics of those objects.
 While firmly footed in this prior history, the LSST software stack has
 largely been written anew, for reasons of performance, extendability, and
-maintainability. All LSST codes have been designed and implemented
+maintainability. All LSST codes are being developed with the intent of
 following software engineering best practices, including modularity, clear definition
 of interfaces, continuous integration,
 utilization of unit testing, a single set of documentation and coding
-standards, and others. The primary implementation language is Python and, where
-necessary for performance reasons, C++\footnote{All components implemented
-in C++ have been wrapped and exposed as Python modules to the rest
-of the system. Python truly is the ``native language of LSST''.}.
+standards, and others. The primary user-facing implementation language is Python and, where
+necessary for performance reasons or when the algorithms require access to complex data structures, C++.
```

Comment:
As Tim says, I asked him to make a similar change in his paper.

The thing is that we decided to make the C++ environment richer than it needs to be strictly for performance.  Saying that the C++ is just for performance makes me think of a very sparse C++ environment wrapping the bare minimum (using something like cython rather than SWIG).

I'm not arguing against the richer C++ environment because I think there are some really nice advantages, but I think it may confuse some in the community when we state that there is a minimum of C++ when it is fully half the code.